components/segmenter/src/line.rs

```diff
@@ -1065,6 +1065,7 @@ impl<'l, 's, Y: LineBreakType<'l, 's>> LineBreakIterator<'l, 's, Y> {
 }
 
 #[derive(Debug)]
+#[non_exhaustive]
 pub struct LineBreakTypeUtf8;
```

Comment:
Question: why are the lint rules like `missing_docs` and `exhaustive_structs` not already being applied here?

Reply:
The types aren't actually public, they're exposed via typedefs.

Reply:
This crate really needs a cleanup but I don't think we have time for that now.

Reply:
In particular the module structure is also messy, with everything at the top level. We should perhaps clean that up.